DQM/GEM/plugins/GEMEfficiencyAnalyzer.cc

```diff
@@ -558,21 +567,29 @@ GEMEfficiencyAnalyzer::StartingState GEMEfficiencyAnalyzer::buildStartingState(
     }
   }
 
-  found &= state.isValid();
-
-  if (MuonHitHelper::isGEM(det_id)) {
-    const GEMDetId start_id{det_id};
+  if (found) {
+    found &= state.isValid();
+  }
 
-    const bool are_same_region = gem_layer.id.region() == start_id.region();
-    const bool are_same_station = gem_layer.id.station() == start_id.station();
-    const bool are_same_layer = gem_layer.id.layer() == start_id.layer();
-    if (are_same_region and are_same_station and are_same_layer) {
-      LogDebug(kLogCategory_)
-          << "The starting detector of the muon propagation is same with the destination. Skip this propagation.";
-      found = false;
-    }
+  if (found and (det_id.det() == DetId::Detector::Muon)) {
+    found &= isMuonSubdetAllowed(det_id, gem_layer.id.station());
   }
 
+  if (found) {
+    if (MuonHitHelper::isGEM(det_id)) {
```

Comment:
Is it acceptable that `found` remains `true` is the detector is not `GEM`?

Reply:
`det_id` is for muon propagation's initial state, and GEM can also be a candidate unless the initial state is on the same layer as the destination surface.